docs/source/index.rst

```diff
@@ -150,6 +150,12 @@ Documentation
    community/meetups
    community/sponsors
 
+.. toctree::
+   :maxdepth: 2
+   :caption: Contributing
+
+   contributing/overview.md
```

Comment:
Suggest moving this to first one inside Developer Documentation

Reply:
@simon-mo I wasn't really sure what the desired scope of that section was. It mostly looks like Python API documentation with a target audience of people using vllm as a library.

and the `Profiling vLLM` page that does seem more aligned to the "contributing" section I was proposing here.

and then there's the `Dockerfile` page that I'm not sure really fits in either one.

in summary, we could have

* API Documentation -- targeted at people using vllm as a library
* Contributing -- targeted at people working on the vLLM source

and I wouldn't use "Developer Documentation" in either place, since that's a bit ambiguous between the two personas.

Finally, the last alternative I see would be to keep contributor information out of the docs site completely and keep it in its own directory to be browsed on github. That's another way to keep separation between the personas.

Reply:
I just pushed another commit with the changes I proposed above.

* move profiling and dockerfile content to the new "Contributing" section
* rename "Developer Documentation" to "API Documentation"

Reply:
I think we should also move the "vLLM Paged Attention" page as well as the "Input Processing Pipeline" and "Adding a Multimodal Plugin" pages to "Contributing".

Reply:
Thanks for highlighting those docs. I think this warrants a third section:

1. `API Documentation` - API reference documents, primarily aimed at those using vLLM as a library
2. `Design` - design docs, discussing how parts of vllm work (more technical)
3. `Contributing` - information on how to work with vllm as a developer and contributor (more policy and process oriented)

Reply:
> Thanks for highlighting those docs. I think this warrants a third section:
> 
> 1. `API Documentation` - API reference documents, primarily aimed at those using vLLM as a library
> 2. `Design` - design docs, discussing how parts of vllm work (more technical)
> 3. `Contributing` - information on how to work with vllm as a developer and contributor (more policy and process oriented)

just finished organizing into these sections